### PR TITLE
Implement LightblueNotificationRepository

### PR DESCRIPTION
**Pull request by alechenninger**
Sans error cases, working on that now in another branch...

### CODE REVIEW

File: lightblue/src/main/java/org/esbtools/eventhandler/lightblue/ByEntityNameNotificationFactory.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+package org.esbtools.eventhandler.lightblue;
+
+import org.esbtools.lightbluenotificationhook.NotificationEntity;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+public class ByEntityNameNotificationFactory implements NotificationFactory {
```
Comment (kahowell) [inline lightblue/src/main/java/org/esbtools/eventhandler/lightblue/ByEntityNameNotificationFactory.java]: Class name is a little non-obvious (to me at least). JavaDoc might be sufficient to correct though.

File: lightblue/src/main/java/org/esbtools/eventhandler/lightblue/FindRequests.java [hunk 2]
```diff
@@ -25,10 +25,23 @@
 import com.redhat.lightblue.client.request.data.DataFindRequest;
 
 import org.esbtools.eventhandler.lightblue.model.DocumentEventEntity;
+import org.esbtools.lightbluenotificationhook.NotificationEntity;
 
 public abstract class FindRequests {
-    public static DataFindRequest newNotificationsForEntitiesUpTo(String[] entities, int maxEvents) {
-        return null;
+    public static DataFindRequest oldestNewNotificationsForEntitiesUpTo(String[] entities,
```
Comment (kahowell) [inline lightblue/src/main/java/org/esbtools/eventhandler/lightblue/FindRequests.java]: Wordy name. Don't have any suggestions for better one though...

Comment (kahowell) [inline lightblue/src/main/java/org/esbtools/eventhandler/lightblue/FindRequests.java]: Actually, maybe just removing "New" would help. Would also make it consistent with terminology in `LightblueNotificationRepository`

Comment (alechenninger) [inline lightblue/src/main/java/org/esbtools/eventhandler/lightblue/FindRequests.java]: That's fine with me. Technically it should be "unprocessed" notifications. The API has the contract of "don't hand me the same thing more than once" so an implementation of this is to use statuses. This request is one that is modelled around said status -- it doesn't just grab oldest notifications, it is specifically oldest unprocessed notifications.

IMHO it's important to be specific with the request names for readability, but readable to me is not the same as readable to everyone else so, if you think it is okay to be less specific I'm okay with that.

File: lightblue/src/main/java/org/esbtools/eventhandler/lightblue/ByEntityNameNotificationFactory.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+/**
+ * Supports creating many types of {@link LightblueNotification}s based on the entity name in the
+ * provided {@link NotificationEntity}.
+ */
+public class ByEntityNameNotificationFactory implements NotificationFactory {
```
Comment (kahowell) [inline lightblue/src/main/java/org/esbtools/eventhandler/lightblue/ByEntityNameNotificationFactory.java]: `NotificationFactoryMapByEntityName`?

File: lightblue/src/test/java/org/esbtools/eventhandler/lightblue/LightblueNotificationRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+            Instant occurrenceDate) {
+        return new StringNotification(value, NotificationEntity.Operation.INSERT, "tester",
+                Clock.fixed(occurrenceDate, ZoneOffset.UTC)).wrappedNotificationEntity();
+    }
+
+    private static NotificationEntity[] randomNotificationEntities(int amount) {
+        NotificationEntity[] entities = new NotificationEntity[amount];
+
+        for (int i = 0; i < amount; i++) {
+            entities[i] = notificationEntityForStringInsert(UUID.randomUUID().toString());
```
Comment (kahowell) [inline lightblue/src/test/java/org/esbtools/eventhandler/lightblue/LightblueNotificationRepositoryTest.java]: I <3 UUIDs

Comment (alechenninger) [inline lightblue/src/test/java/org/esbtools/eventhandler/lightblue/LightblueNotificationRepositoryTest.java]: ;) I thought you'd like that haha

**Comment by kahowell** [comment]
LGTM. Merge soon?

**Pull request merged**